includes/class-wc-product-variable.php

```diff
@@ -282,9 +282,10 @@ public function get_virtual( $context = 'view' ) {
 	/**
 	 * Get an array of available variations for the current product.
 	 *
+	 * @param bool $render_variations Allows avoiding performance bottlenecks caused by get_available_variation().
```

Comment:
```suggestion
	 * @param bool $render_variations Prepares a data array for each variant for output in the add to cart form. Pass `false` to only return the available variations as objects.
```

Reply:
Why only for the add to cart form? It is used in a lot more cases? I think we could skip the first sentence, if we want to keep it straightforward, like you first suggested. Agree?

Reply:
> Why only for the add to cart form?

That is the single use-case that is documented for get_available_variation():
https://github.com/netzstrategen/woocommerce/blob/0b92b293da582302b7467c784bec86e65ed3d5d9/includes/class-wc-product-variable.php#L326-L333

It explains to the reader why generating this data would make sense for template rendering (for that limited use-case), so in my opinion it is important information to keep.